hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterNetworkTopologyServlet.java

```diff
@@ -49,9 +51,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     }
 
     Router router = RouterHttpServer.getRouterFromContext(context);
-    DatanodeInfo[] datanodeReport =
-        router.getRpcServer().getDatanodeReport(
-            HdfsConstants.DatanodeReportType.ALL);
+    DatanodeInfo[] datanodeReport = null;
+    if (router.getRpcServer().isAsync()) {
+      router.getRpcServer().getDatanodeReportAsync(
+          HdfsConstants.DatanodeReportType.ALL, true, 0);
+      try {
+        datanodeReport = syncReturn(DatanodeInfo[].class);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
```

Comment:
```java
      } catch (IOException e) {
        throw e;
      } catch (Exception e) {
        throw new IOException(e);
      }
```

Reply:
@KeeProMise Thanks for reviewing, have fixed~